website/docs/error-reference.md

```diff
@@ -1515,16 +2114,114 @@ class C
 end
 ```
 
+## 5049
+
+Parameters given type annotations in a method signature must be provided in the
+same order they appear in the method definition, even for keyword parameters for
+which order would not usually be required. This is for readability and
+consistency.
+
+Also, all optional keyword parameters must come after all required keyword
+parameters.
+
+## 5050
+
+Sealed classes (or modules) can only be inherited (or included/extended) in the
+same file as the parent.
+
+For more information, see [Sealed Classes](sealed.md)
+
+## 5051
+
+See [Override Checking](override-checking.md).
+
+## 5052
+
+When providing bounds for a type member with `lower` and `upper`, the lower type
+bound must be a subtype of (or equivalent to) the `upper` type bound.
+
+Otherwise, it would never be possible to instantiate the generic type because
+the bounds would never be satisfiable.
+
+## 5053
+
+For classes that subclass generic classes, the child's lower and upper bound
+must still be within the lower and upper bound range specified by the parent
+class.
+
+This problem can come up from time to time when the parent class uses `fixed`
+when it meant to use `upper`. For example:
+
+```ruby
+class IntOrStringBox
+  extend T::Generic
+  Elem = type_member {{fixed: T.any(Integer, String)}}
+end
+
+class IntBox < IntOrStringBox
+  Elem = type_member {{fixed: Integer}} # error, incompatible bound
+end
+```
+
+In this case we'd like `IntBox` to be a subclass of `IntOrString` box, but since
+we're using `fixed` Sorbet prevents us.
+
+A fix is to use only `upper` instead:
+
+```ruby
+class IntOrStringBox
+  extend T::Generic
+  Elem = type_member {{upper: T.any(Integer, String)}}
+end
+
+class IntBox < IntOrStringBox
+  Elem = type_member {{upper: Integer}}
+end
+```
+
+But this does come at the slight cost that now all type annotations that used to
+be using `IntOrStringBox` without any generic type arguments must now provide
+one:
+
+```ruby
+T.let(IntBox[Integer].new, IntOrStringBox) # error: Generic class without type arguments
+```
+
+The fix is to change all occurrences of `IntOrStringBox` without type arguments
+to specify `IntOrStringBox[T.any(Integer, String)]`.
+
 ## 5054
 
 Use of `implementation` has been replaced by `override`.
 
-## 5056
+## 5055
+
+Sorbet does not consider a class "fully resolved" until all of its type members
+have also been fully resolved. Sorbet cannot fully resolve a type member in a
+class until all constants mentioned in its right-hand side have been fully
+resolved. So if any of those right-hand-side constants are themselves generic
```

Comment:
"Right-hand{ ,-}side feels like a little too much Sorbet internals talk, but I'm not sure what an equally descriptive term would be in a user context.

Reply:
maybe? it's hopefully clear from context what the right and left hand side of a type member declaration is. the first time someone asks about it i'm happy to think of an alternate phrasing.